staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -1213,11 +1213,15 @@ func expandIfFilePattern(pattern string) ([]string, error) {
 	if _, err := os.Stat(pattern); os.IsNotExist(err) {
 		matches, err := filepath.Glob(pattern)
 		if err == nil && len(matches) == 0 {
-			return nil, fmt.Errorf("no match")
+			return nil, fmt.Errorf("the path %q does not exist", pattern)
+		}
+		if err == filepath.ErrBadPattern {
+			return nil, fmt.Errorf("pattern %q is not valid: %v", pattern, err)
 		}
-		if err == nil || err != filepath.ErrBadPattern {
+		if err != nil {
 			return matches, err
 		}
+		return matches, nil
```

Comment:
oh, I guess this can collapse to `return matches, err`

Reply:
Fixed, I think I should call it a day at this point in time :wink: